tailcfg/tailcfg.go

```diff
@@ -192,19 +190,74 @@ type Node struct {
 	MachineAuthorized bool `json:",omitempty"` // TODO(crawshaw): replace with MachineStatus
 }
 
-// DefaultDisplayName returns a value suitable
-// for using as the default value for n.DisplayName.
-func (n *Node) DefaultDisplayName() string {
-	if n.Name != "" {
-		// Use the Magic DNS prefix as the default display name.
-		return dnsname.ToBaseName(n.Name)
+// displayName holds the various names that can be used for this node in UIs.
+type displayName struct {
+	Name          string // primary name for the node; populated with the node's magic dns name when possible
```

Comment:
// MagicDNS base name (for normal non-shared-in nodes) , FQDN (without trailing dot, for shared-in nodes), or Hostname (if no MagicDNS)